src/ng/compile.js

```diff
@@ -1587,6 +1587,84 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     return cssClassDirectivesEnabledConfig;
   };
 
+
+  /**
+   * The security context of DOM Properties.
+   * @private
+   */
+  var PROP_CONTEXTS = Object.create(null);
+
+  /**
+   * @ngdoc method
+   * @name $compileProvider#addPropertyContext
+   * @description
+   *
+   * Defines the security context for HTML properties bound by ng-prop-*
+   * 
+   * @param {string} the context type
+   * @param {string} the element name or '*' to match any element
+   * @param {string} the property name
+   */
+  this.addPropertyContext = function(ctx, elementName, propertyName) {
+    PROP_CONTEXTS[(elementName.toLowerCase() + '|' + propertyName.toLowerCase())] = ctx;
+  };
+
+  /* Default property contexts.
+   *
+   * Copy of https://github.com/angular/angular/blob/6.0.6/packages/compiler/src/schema/dom_security_schema.ts#L31-L58
+   * Changing:
+   * - SecurityContext.* => $sce.*
+   * - STYLE => CSS
+   * - various URL => MEDIA_URL
+   */
+  (function setupPropertyContexts() {
+    function registerContext(ctx, values) {
+      forEach(values, function(v) { PROP_CONTEXTS[v.toLowerCase()] = ctx; });
+    }
+
+    registerContext(SCE_CONTEXTS.HTML, [
+      'iframe|srcdoc',
+      '*|innerHTML',
+      '*|outerHTML',
+    ]);
+    registerContext(SCE_CONTEXTS.CSS, ['*|style']);
+    registerContext(SCE_CONTEXTS.URL, [
+      '*|formAction',
+      'area|href',       'area|ping',
+      'a|href',          'a|ping',
+      'blockquote|cite',
+      'body|background',
+      'del|cite',
+      'form|action',
+      'input|src',
+      'ins|cite',
+      'q|cite',
+      'script|src',
+      'video|poster'
+    ]);
+    registerContext(SCE_CONTEXTS.MEDIA_URL, [
+      'audio|src',
+      'img|src',    'img|srcset',
+      'source|src', 'source|srcset',
+      'track|src',
+      'video|src'
+    ]);
+    registerContext(SCE_CONTEXTS.RESOURCE_URL, [
+      'applet|code',      'applet|codebase',
+      'base|href',
+      'embed|src',
+      'frame|src',
+      'head|profile',
+      'html|manifest',
+      'iframe|src',
+      'link|href',
+      'media|src',
+      'object|codebase',  'object|data',
+      'script|src',
+    ]);
+  })();
```

Comment:
I'd say since we already have attribute sanitizing built in we should probably do the same for properties?

Reply:
I wonder how much this affects the overall size. Since these are strings, they can't be minified, but there are less of them than I thought there would be and they probably compress fine. So, I don't think this is a problem, but am still curious about the size increase.

Reply:
As a quick (and inaccurate) test, this setup method on it's own is 1.2kb and 0.7kb minfied. Should minify a bit better when not isolated and then gzip would help a bit more...

Reply:
Size is a bit of a concern, especially because there might be very limited take up of this feature.
What if we put the core changes into angular.js but always throw an exception if the `ngProp` module is not loaded. Then we could move as much of this sanitization (and anything else auxiliary) into that module and keep the core changes to a minimum?

Reply:
I'd really prefer if it was usable without adding another file/module, especially since that file will just be this property list... all the actual functionality is baked into `$compile` and will already be there. Although I guess the `ng-on-*` is the one that will be used more, so that one is more important to me to be baked in.

Is it mainly just the size?

I could probably make this a bit smaller (concatenate all the strings instead of arrays of strings...).

Reply:
I mentioned I'd try to make this minify better... looks like the closure compiler already does what I was thinking of (convert `['a', 'b', 'c']` to `'a b c'.split(' ')`).

When minifying only this IIFE it is 371 bytes gzipped. When minifying the full library, then measuring just this IIFE it is 348 bytes gzipped (however it might have been minified to gzip better along with the rest of the app?).

So basically ~350 bytes is a good estimate.